src/safeposix/dispatcher.rs

```diff
@@ -141,14 +146,30 @@ macro_rules! check_and_dispatch_socketpair {
 // to increase I/O performance by bypassing the dispatcher and type checker
 #[no_mangle]
 pub extern "C" fn quick_write(fd: i32, buf: *const u8, count: usize, cageid: u64) -> i32 {
-  unsafe { CAGE_TABLE[cageid as usize].as_ref().unwrap().write_syscall(fd, buf, count) }
+    interface::check_cageid(cageid);
+    unsafe { CAGE_TABLE[cageid as usize].as_ref().unwrap().write_syscall(fd, buf, count) }
 }
 
 #[no_mangle]
 pub extern "C" fn quick_read(fd: i32, buf: *mut u8, size: usize, cageid: u64) -> i32 {
+    interface::check_cageid(cageid);
     unsafe { CAGE_TABLE[cageid as usize].as_ref().unwrap().read_syscall(fd, buf, size) }
 }
```

Comment:
The call to **interface::check_cageid(cageid)** within both the **quick_write** and **quick_read** functions serves as a runtime check to validate the provided cageid. However, this introduces additional runtime overhead. Given that both **writing** and **reading** operations are frequently called during execution, the cumulative overhead may be substantial.